src/main/java/seedu/address/model/person/PersonMatchesQueryPredicate.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Name}, {@code Tags} or {@code Assets} matches the search query.
+ * The algorithm removes all whitespaces and checks if the query is a substring of a field.
+ * The substring matching is case-insensitive.
+ */
+public class PersonMatchesQueryPredicate implements Predicate<Person> {
+
+    private static final String whitespaceRegex = "\\s+";
+    private static final String emptyString = "";
```

Comment:
Not sure but these seem like constants. Should it be in SCREAMING_SNAKE_CASE?

Reply:
Yes, changed.